src/components/Dropdown/DropdownList.tsx

```diff
@@ -56,21 +57,98 @@ export const DropdownList = (props: IDropdownListProps): ReactElement => {
     ],
   });
 
+  const close = () => {
```

Comment:
This is okay, however when you're essentially "watching" for changes to a state variable (in this case `visible`) its best to do it explicitly so you always stay up-to-date.  I would move the `onClosed()` and `onExpanded()` into a `useEffect` that watches for changes to `visible` so they stay in sync.
For example:

```js
// call our callbacks when visible changes
useEffect(() => (visible ? onExpanded() : onClosed()), [visible]);

// helpers for setting visibility
const close = () => visible && setVisible(false);
const open = () => !visible && setVisible(true);

/* Click on dropdown */
const handleClick = () => {
  setVisible(!visible);
};
```

Reply:
Updated